### PR TITLE
Add Release Gene to Curation Page Tools Tab

### PR DESCRIPTION
**Pull request by bprize15**
resolves https://github.com/oncokb/oncokb/issues/3651

### CODE REVIEW

File: src/main/webapp/app/components/tabs/CurationToolsTab.tsx [hunk 1]
```diff
@@ -1,105 +1,142 @@
...
+  }
+
+  useEffect(() => {
+    const geneData = geneEntities.find(entity => entity.hugoSymbol === gene.name);
+    setIsReleased(geneData?.flags?.some(flag => isReleasedFlag(flag)) || false);
+    geneToUpdate.current = geneData;
+  }, []);
+
+  function getStatusIcon(checkPassed: boolean) {
+    return checkPassed ? <FaRegCheckCircle className="text-success mt-1 mr-2" /> : <FaRegCircleXmark className="text-danger mt-1 mr-2" />;
```
Comment (zhx828) [inline src/main/webapp/app/components/tabs/CurationToolsTab.tsx:48]: Let's move the common class names as a constant within the method.

File: src/main/webapp/app/components/tabs/CurationToolsTab.tsx [hunk 2]
```diff
@@ -1,105 +1,142 @@
...
+    return flag.type === 'GENE_PANEL' && flag.flag === 'ONCOKB';
+  }
+
+  useEffect(() => {
+    const geneData = geneEntities.find(entity => entity.hugoSymbol === gene.name);
+    setIsReleased(geneData?.flags?.some(flag => isReleasedFlag(flag)) || false);
+    geneToUpdate.current = geneData;
+  }, []);
+
+  function getStatusIcon(checkPassed: boolean) {
```
Comment (zhx828) [inline src/main/webapp/app/components/tabs/CurationToolsTab.tsx:47]: Given all use cases within this file, let's rename the pass to checkNotPassed and update the usage.

File: src/main/webapp/app/components/tabs/CurationToolsTab.tsx [hunk 3]
```diff
@@ -1,105 +1,142 @@
...
+            Gene summary is not empty
+          </Row>
+          <Row className="mb-1">
+            {getStatusIcon(!geneBackgroundEmpty)}
+            Gene background is not empty
+          </Row>
+          <Row className="mb-3">
+            {getStatusIcon(!geneTypeEmpty)}
+            Gene type is specified
+          </Row>
```
Comment (zhx828) [inline src/main/webapp/app/components/tabs/CurationToolsTab.tsx:105]: Can we aggregate these checks into a list? The `Row` here can be populated based on the list. Also `confirmButtonDisabled` can be calculated based on the list. 
Test needs to be differentiated with `optional` and `requried`

**Review by zhx828 — CHANGES_REQUESTED** [review]
- Could you also update the title to reflect the purpose of the PR? Also make sure capitalize the First word so it can be used in the release note. We don't generate release note yet in this repo, but good to have it standardized.
- Could you add label, `feature`?

File: src/main/webapp/app/components/tabs/CurationToolsTab.tsx (hunk 2 above)
Comment (bprize15) [reply src/main/webapp/app/components/tabs/CurationToolsTab.tsx:47]: After defining the tests to be run when creating a gene, we now have a `passed` variable for each test. The use cases all now use `getStatusIcon(test.passed)`, so will leave the same

File: src/main/webapp/app/components/tabs/CurationToolsTab.tsx [hunk 4]
```diff
@@ -1,105 +1,159 @@
...
+    },
+    {
+      passed: !!gene.background,
+      text: 'Gene background is not empty',
+      type: 'required',
+    },
+    {
+      passed: !!gene.type.ocg || !!gene.type.tsg,
+      text: 'Gene type is specified',
+      type: 'required',
```
Comment (zhx828) [inline src/main/webapp/app/components/tabs/CurationToolsTab.tsx:40]: Forgot to mention, the gene type check is optional.

File: src/main/webapp/app/components/tabs/CurationToolsTab.tsx [hunk 5]
```diff
@@ -1,105 +1,159 @@
...
+  text: string;
+  type: 'optional' | 'required';
+};
+
+export function CurationToolsTab({ metaList, addMetaListListener, gene, geneEntities, searchGenes, updateGene, searchFlags }: StoreProps) {
+  const reviewList = metaList?.[gene.name].review;
+
+  const tests: ReleaseGeneTestData[] = [
+    {
+      passed: reviewList ? !Object.keys(reviewList).some(key => key !== 'currentReviewer' && reviewList[key]) : true,
```
Comment (zhx828) [inline src/main/webapp/app/components/tabs/CurationToolsTab.tsx:23]: Can we create a constant of `currentReviewer`? @calvinlu3 this is already related to ur work I believe on how to determine whether there is review content.

**Review by zhx828 — COMMENTED** [review]
Two last comments, once fixed, you can merge. Thanks!

**Pull request merged**